packages/crypto-wallet-core/src/derivation/btc/index.ts

```diff
@@ -3,20 +3,24 @@ import { IDeriver } from '..';
 export abstract class AbstractBitcoreLibDeriver implements IDeriver {
   public abstract bitcoreLib;
 
-  deriveAddress(network, pubKey, addressIndex, isChange) {
+  deriveAddress(network, pubKey, addressIndex, options) {
+    const { isChange, isBech32 } = options;
     const xpub = new this.bitcoreLib.HDPublicKey(pubKey, network);
     const changeNum = isChange ? 1 : 0;
+    const type = isBech32 ? 'witnesspubkeyhash' : 'pubkeyhash';
     const path = `m/${changeNum}/${addressIndex}`;
-    return this.bitcoreLib.Address(xpub.derive(path).publicKey, network).toString();
+    return this.bitcoreLib.Address(xpub.derive(path).publicKey, network, type).toString();
   }
 
-  derivePrivateKey(network, xPriv, addressIndex, isChange) {
+  derivePrivateKey(network, xPriv, addressIndex, options) {
+    const { isChange, isBech32 } = options;
     const xpriv = new this.bitcoreLib.HDPrivateKey(xPriv, network);
     const changeNum = isChange ? 1 : 0;
+    const type = isBech32 ? 'witnesspubkeyhash' : 'pubkeyhash';
```

Comment:
If the address is change, it should always be bech32. The user will not see the address since it's internal, but will save on fees because it is using segwit.